packages/kit/src/runtime/server/page/fetch.js

```diff
@@ -176,6 +175,25 @@ export function create_fetch({ event, options, state, route, prerender_default }
 
 			const external_request = new Request(requested, /** @type {RequestInit} */ (opts));
 			response = await options.hooks.externalFetch.call(null, external_request);
+
+			if (opts.mode === 'no-cors') {
+				response = new Response('', {
+					status: response.status,
+					statusText: response.statusText,
+					headers: response.headers
+				});
+			} else {
+				if (url.origin !== event.url.origin) {
```

Comment:
@Rich-Harris, if url.origin = 'http://api.app.domain.io and event.url.origin = 'http://app.domain.io' this will result in checking the cors, even if this is not considered as a cross-origin request. Or am I wrong?